File: src/types/dto/workspaces.ts
export type WorkspaceDto = {
  id: string;
  projectId: string;
  cardBoard: string | null;
  cardId: string | null;
  name: string;
  repoPath: string;
  worktreePath: string;
  branch: string;
  status: string;
  archived: boolean;
  createdAtMs: number;
  updatedAtMs: number;
};

export type AgentSessionDto = {
  id: string;
  workspaceId: string;
  executor: string;
  ptySessionId: string;
  status: string;
  lastPrompt: string | null;
  createdAtMs: number;
  updatedAtMs: number;
};

export type WorkspaceChangeEntry = {
  path: string;
  status: string;
  additions: number;
  deletions: number;
};

import type { GitFileDiffDto } from "./git";

export type { GitFileDiffDto };

export type WorkspaceRepoInfoDto = {
  branch: string;
  base: string;
  owner: string | null;
  repo: string | null;
  pushed: boolean;
};

export type CreatedPrDto = { number: number; htmlUrl: string };

export type PrStatusDto = {
  number: number;
  htmlUrl: string;
  state: string;
  merged: boolean;
  mergeable: boolean | null;
  head: string;
  base: string;
};

export type WorkspaceDetailDto = {
  workspace: WorkspaceDto;
  sessions: AgentSessionDto[];
  changes: WorkspaceChangeEntry[];
};

export type ExecutionViewDto = {
  session: AgentSessionDto;
  liveStatus: string;
  exitCode: number | null;
  outputTail: string;
};

export type ExecutorInfoDto = { id: string; display: string; available: boolean };

File: src-tauri/src/models/workspaces.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceDto {
    pub id: String,
    pub project_id: String,
    pub card_board: Option<String>,
    pub card_id: Option<String>,
    pub name: String,
    pub repo_path: String,
    pub worktree_path: String,
    pub branch: String,
    pub status: String,
    pub archived: bool,
    pub created_at_ms: i64,
    pub updated_at_ms: i64,
}

/// One coding-agent run inside a workspace (PTY-backed, like task sessions).

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentSessionDto {
    pub id: String,
    pub workspace_id: String,
    pub executor: String,
    pub pty_session_id: String,
    pub status: String,
    pub last_prompt: Option<String>,
    pub created_at_ms: i64,
    pub updated_at_ms: i64,
}


